File: tests/fixtures/v2_phase_io/go_api/verification/input/internal/handlers/health_handler.go
package handlers

import (
	"encoding/json"
	"net/http"
)

// HealthHandler handles GET /healthz requests.
type HealthHandler struct{}

// NewHealthHandler creates a new HealthHandler.
func NewHealthHandler() *HealthHandler {
	return &HealthHandler{}
}

// ServeHTTP implements http.Handler for the health check endpoint.
// Returns HTTP 200 with body {"status":"ok"} and Content-Type application/json.
func (h *HealthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodGet {
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusMethodNotAllowed)
		json.NewEncoder(w).Encode(map[string]string{"error": "method not allowed"})
		return
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	json.NewEncoder(w).Encode(map[string]string{"status": "ok"})
}

File: tests/fixtures/v2_phase_io/typescript_express_api/verification/input/src/routes/health_handler.ts
// Health Handler — 处理 GET /healthz 端点逻辑

import { Request, Response, Router } from 'express';

/**
 * Handles the GET /healthz endpoint.
 * Returns a simple health check response with HTTP 200.
 */
export function healthHandler(_req: Request, res: Response): void {
  res.status(200).json({ status: 'ok' });
}

/**
 * Registers the health check route on the given router.
 * @param router - Express Router instance
 */
export function registerHealthRoute(router: Router): void {
  router.get('/healthz', healthHandler);
}
